### PR TITLE
added host page transition changes/notifications

### PR DESCRIPTION
**Pull request by zzulanas**
### 📃 Summary
Added code to the event listeners page to dispatch URL Changes and emit the events to the socket only if user is host. My spaghetti code is pretty minimal, but may need some prettifying or changes.

### 🔍 What's the new behavior?
Now the party URL will update along with a change in videos on in-page navigation, along with sending a socket event to the server.

### 🏷 Task Link
[Story link on Notiono](https://www.notion.so/Handle-Page-Transition-Events-f4a65c8f34f9419aae500dea7c800fd0)

### ↩️ Depends On

Link any other PRs here.

### 🧪 QA

- List the steps to test the changes here.
- Run the extension
- Create a party 
- Navigate from one video to another, check that URL in party popup changes accordingly

### 🛑 Risk Analysis

- [ ] Does it update dependencies?

### CODE REVIEW

**Review by RyanMatthewJacobs — APPROVED** [review]

**Review by stnichol — COMMENTED** [review]
Looks good Zach! What is UPDATE_PARTY used for? Is it for a later feature? Also is line 21 of listeners/index.ts necessary if the roomID doesn't change? Thanks!

**Comment by zzulanas** [comment]
UPDATE_PARTY was just an event idea I was trying to implement on the create_party() function for when the room changes, I reverted some of the changes but I forgot I left the events in there, I was gonna try and use it for this sprint when changing URLs, but we can easily get rid of it

**Review by stnichol — APPROVED** [review]

File: src/content-script/listeners/index.ts
Comment (rkrishn7) [inline src/content-script/listeners/index.ts:26]: - Can we export into a function. There are more cases we might need to handle (socket teardown if not host) and we don't want all the logic to live in this block.
- Can you use `query-string` to generate the new join_url? See https://github.com/rkrishn7/playback_sync_api/blob/main/src/lib/services/party.ts#L16

File: src/content-script/reducers/party.ts
Comment (rkrishn7) [inline src/content-script/reducers/party.ts:15]: Reminder to revert back

File: src/content-script/listeners/index.ts
Comment (rkrishn7) [inline src/content-script/listeners/index.ts:89]: Remove pls

File: src/lib/constants/party.ts
Comment (rkrishn7) [inline src/lib/constants/party.ts:7]: If this isn't being used right now, let's remove and only add when needed.

File: src/lib/constants/socket.ts
Comment (rkrishn7) [inline src/lib/constants/socket.ts:5]: Same, remove if not being used.

**Review by rkrishn7 — COMMENTED** [review]
Looks good. Some feedback provided above.

Going forward, I think we should only merge changes when they add functionality or meet the requirements of the ticket/story. You've got the base here, but there will need to be server code written to handle the events.

It's fine if you want to define them in two separate tickets, but let's wait on the feature to be implemented before we merge into the production branch.

**Comment by zzulanas** [comment]
> Looks good. Some feedback provided above.
> 
> Going forward, I think we should only merge changes when they add functionality or meet the requirements of the ticket/story. You've got the base here, but there will need to be server code written to handle the events.
> 
> It's fine if you want to define them in two separate tickets, but let's wait on the feature to be implemented completely before we merge into the production branch.

I've got the server changes on my local machine, I don't think I have access to make a PR yet, or at least last time I checked

**Comment by rkrishn7** [comment]
> > Looks good. Some feedback provided above.
> > Going forward, I think we should only merge changes when they add functionality or meet the requirements of the ticket/story. You've got the base here, but there will need to be server code written to handle the events.
> > It's fine if you want to define them in two separate tickets, but let's wait on the feature to be implemented completely before we merge into the production branch.
> 
> I've got the server changes on my local machine, I don't think I have access to make a PR yet, or at least last time I checked

It looks like you need to accept the invite.

**Pull request merged**
